common/djangoapps/microsite_configuration/tests/test_backends.py

```diff
@@ -0,0 +1,428 @@
+            microsite.get_all_orgs(),
+            set(self.microsite.get_orgs())
+        )
+
+    def test_clear(self):
+        """
+        Tests microsite.clear works as expected.
+        """
+        microsite.set_by_domain(self.microsite.subdomain)
+        self.assertEqual(
+            microsite.get_value('platform_name'),
+            self.microsite.values['platform_name']
+        )
+        microsite.clear()
+        self.assertIsNone(microsite.get_value('platform_name'))
+
+    def test_enable_microsites(self):
+        """
+        Tests microsite.enable_microsites works as expected.
+        """
+        # remove microsite root directory paths first
+        settings.DEFAULT_TEMPLATE_ENGINE['DIRS'] = [
+            path for path in settings.DEFAULT_TEMPLATE_ENGINE['DIRS']
+            if path != settings.MICROSITE_ROOT_DIR
+        ]
+        with patch.dict('django.conf.settings.FEATURES', {'USE_MICROSITES': False}):
+            microsite.enable_microsites(log)
+            self.assertNotIn(settings.MICROSITE_ROOT_DIR, settings.DEFAULT_TEMPLATE_ENGINE['DIRS'])
+        with patch.dict('django.conf.settings.FEATURES', {'USE_MICROSITES': True}):
+            microsite.enable_microsites(log)
+            self.assertIn(settings.MICROSITE_ROOT_DIR, settings.DEFAULT_TEMPLATE_ENGINE['DIRS'])
+
+    def test_get_all_configs(self):
+        """
+        Tests microsite.get_all_config works as expected.
+        """
+        microsite.set_by_domain(self.microsite.subdomain)
+        configs = microsite.get_all_config()
+        self.assertEqual(len(configs.keys()), 1)
+        self.assertEqual(configs[self.microsite.key], self.microsite.values)
+
+    def test_set_config_by_domain(self):
+        """
+        Tests microsite.set_config_by_domain works as expected.
+        """
+        microsite.clear()
+        # if microsite config does not exist
+        microsite.set_by_domain('unknown')
+        self.assertIsNone(microsite.get_value('platform_name'))
+
+        # if no microsite exists
+        Microsite.objects.all().delete()
+        microsite.clear()
+        microsite.set_by_domain('unknown')
+        self.assertIsNone(microsite.get_value('platform_name'))
+
+        # if microsite site has no organization it should raise exception
+        new_microsite = MicrositeFactory.create()
+        new_microsite.key = 'test_microsite'
+        new_microsite.subdomain = 'microsite.test'
+        # This would update microsite so we test MicrositeHistory has old microsite
+        new_microsite.save()
+        self.assertEqual(MicrositeHistory.objects.all().count(), 2)
+        with self.assertRaises(Exception):
+            microsite.set_by_domain('microsite.test')
+
+
+@patch(
+    'microsite_configuration.microsite.BACKEND',
+    microsite.get_backend(
+        'microsite_configuration.backends.filebased.SettingsFileMicrositeBackend', AbstractBaseMicrositeBackend
+    )
+)
+class FilebasedMicrositeBackendTests(TestCase):
```

Comment:
We'll need to rename this test class to match whatever we choose for the actual class